project/frontend/src/components/ResultsMap.js

```diff
@@ -1,12 +1,54 @@
-import React from 'react';
+import React, { Fragment, useState } from 'react';
 import GoogleMapReact from 'google-map-react';
 import Lunch from '../assets/lunch.svg';
 
 function MapContainer(props) {
   const restaurants = props.restaurants;
   const userCenter = props.userLocation;
+  const [activeMarker, setActiveMarker] = useState({});
+  const [showInfoWindows, setShowInfoWindows] = useState([
+    false,
+    false,
+    false,
+    false,
```

Comment:
for this, I think we should add the param name for the state to make it clear what the state is referring to 
name1: false,
name2: false 

etc